test/sub.mk

```diff
@@ -22,26 +32,29 @@ all:
 
 compile: compile.stamp
 
-compile.stamp: ../../../compiler/testSuite/$(tdir).balt
+ifeq ($(bal_files),)
+# sub dir only contains e cases
+compile.stamp:
+	@touch $@
+else
+compile.stamp: $(bal_files)
 	-rm -fr llnew
-	-rm -fr expectnew
-	$(JAVA) -jar $(COMPILER_JAR) --outDir llnew --expectOutDir expectnew $^
+	mkdir -p llnew
+	$(JAVA) -jar $(COMPILER_JAR) --outDir llnew $?
 	mkdir -p ll
-	mkdir -p expect
-	-../../update.sh expectnew expect txt
-	-../../update.sh llnew ll ll
+	cd llnew; for f in *.ll; do cmp -s $$f ../ll/$$f || mv $$f ../ll/; done
 	-rm -fr llnew
-	-rm -fr expectnew
 	@touch $@
+endif
 
+# This compiles, runs and checks the output of ll/*.ll
 testll: fail.txt
```

Comment:
This isn't showing the changes relative to the current Makefile.  In the current Makefile, we have:

```
testll: out/result/fail.txt $(expect_files) $(exe_files)
```

Did you intend to remove the dependency on expect_files and exe_files? If so, why?

Reply:
You may view the correct diff by selecting just the relevant commit ([Modify makefile content into sub.mk](https://github.com/ballerina-platform/nballerina/pull/424/commits/273d57d46e9d473fdd841e2cee82aa5c6ca85d11)).

Yes it was intentional, I though since fail.txt depends on diff_files and each diff is depended on exe and the expect text, explicitly adding it here is redundant.

Reply:
I see your reasoning, but I am not confident that's a good change, because there are a lot of complexities about how pattern rules work when dependencies do not exist.

In any case, this is an unrelated change and so should be not be mixed in with this.

Reply:
I'll add it back.

Reply:
[Added](https://github.com/ballerina-platform/nballerina/pull/424/commits/6c7e43661dbf49b8e07b8e5349d51a6992f2e64d).